### ISSUE TITLE
Input images

### ISSUE THREAD

**Issue by guist** [manycore-research/PlankAssembly#5]
Hello,

I could run all steps with your data from the preparation until the visualization, and it looks very nice.

However, I still don't understand where to start with my own images which are raster images in jpg format. In your branch dedicated to raster images, I only see svg or json files but not the original images.

Could you please provide me a few precisions on this topic?

Thanks!

Best,
Guillaume

**Comment by bertjiazheng** [comment]
We convert SVG files to raster images for training using the code available at https://github.com/manycore-research/PlankAssembly/blob/atlas/plankassembly/data.py#L200.

Best,
Jia

**Comment by guist** [comment]
Sorry, I think my explanations were not clear. I already have raster images, I don't need to generate them from svg files.

My question was: how to use raster images as input to your testing pipeline?

**Comment by bertjiazheng** [comment]
You should pay attention to the **_intrinsics_** and **_pose_** parameters. 

- For orthographic projection, we set the intrinsic matrix as [here](https://github.com/manycore-research/PlankAssembly/blob/atlas/plankassembly/data.py#L206-L209).
- We set the poses of the front, top, right view in [here](https://github.com/manycore-research/PlankAssembly/blob/atlas/plankassembly/data.py#L185-L187).

Best,
Jia

**Comment by bertjiazheng** [comment]
Feel free to reopen this issue if you still have any questions.

**Comment by guist** [comment]
Hello,

Sorry for the late answer.

Thanks for your precisions. I tried to modify the code as much as possible but unfortunately I have not managed to make it run with my own data. Therefore I am going to try with a different model.

Thanks again for your help!

Best,
Guillaume

**Comment by SanketDhuri** [comment]
how can I use SVG files (of  top ,bottom and side view) as input or just convert this SVG images model required format ie. array

**Comment by DiamondGlassDrill** [comment]
@guist what code did u find that can handle it the same way.
Thanks for support.

**Comment by guist** [comment]
I just tried the code available on the GitHub but it did not work.

In the end, I could not find any other code / model that could solve this problem. Sorry!

**Comment by bertjiazheng** [comment]
Hi @guist @DiamondGlassDrill @SanketDhuri ,

If you are still working on a related topic, here is a quick update from our recent work [CAD2Program](https://manycore-research.github.io/CAD2Program/). Our recent work found that a basic Vision Transformer can understand an engineering drawing (manually arranging the three axis-aligned views on a fixed-size canvas) and perform results comparable to those of the model with vectorized input. Please see the [ViT branch](https://github.com/manycore-research/PlankAssembly/tree/vit) for more information. 

Best,
Jia

**Comment by DiamondGlassDrill** [comment]
Thanks Jia @bertjiazheng cool work!!!
Still of full interest. :)
Need to dive into the code later this week. 
Is the ViT branch the full code base so it should work right with the ViT replaced? The weights of vit-checkpoint_999-precision=0.933-recall=0.918-f1=0.924 is it the right one or is that from PlanckAssembly the original? I guess the old one as the ViTTiny has only ~83MB in size compared to this one.
Thanks for support. Really amazing work!!!!

**Comment by bertjiazheng** [comment]
Yes, the checkpoint is correct. The checkpoint is a PyTorch Lightning checkpoint, including model state, optimizer states, and more.

**Comment by DiamondGlassDrill** [comment]
@bertjiazheng awesome will do some tests this week and let you know. Thanks for the work!
